app/dashboard/page.tsx

```diff
@@ -0,0 +1,34 @@
+'use client';
+
+import { AuthProvider, useAuth } from '../utils/AuthProvider'; // Ensure the path is correct
+
+export default function DashboardPage() {
+  return (
+    <AuthProvider>
+      <Dashboard />
+    </AuthProvider>
+  );
+}
+
+function Dashboard() {
+  const { authUser, isLoggedIn, signOut } = useAuth();
+
+  if (!authUser) {
+    return <p>Loading...</p>;
+  }
+
+  return (
+    <>
+      <header>
+        <h1>Dashboard</h1>
+      </header>
+
+      <main>
+        <p>User is currently: {isLoggedIn ? 'Logged In' : 'Logged Out'}</p>
+        {authUser && <p>User name: {authUser.email}</p>}{' '}
+        {/* Display user's email */}
+        <button onClick={signOut}>Log Out</button> {/* Logout button */}
```

Comment:
Nitpick: it's better to have separate pages/flows for logged in/out users. However, this page looks like it'll be changed in the future, so keep this in mind when/if it does change.